forgerock-openbanking-uk-aspsp-rs/forgerock-openbanking-uk-aspsp-rs-mock-store/forgerock-openbanking-uk-aspsp-rs-mock-store-server/src/main/java/com/forgerock/openbanking/aspsp/rs/store/repository/migration/legacy/payments/FRDomesticConsent.java

```diff
@@ -83,22 +69,20 @@
 
     public OBVersion obVersion;
 
-    public void setPisp(Tpp tpp) {
-        this.pispId = tpp.getId();
-        this.pispName = tpp.getOfficialName();
-    }
-
-    public OBDomestic2 getInitiation() {
-        return domesticConsent.getData().getInitiation();
-    }
-
-    public OBRisk1 getRisk() {
-        return domesticConsent.getRisk();
+    @Override
+    public String getDocumentId() {
+        return this.id;
     }
 
     @Override
-    public boolean isNew() {
-        return created == null;
+    public String getCountrySubDivision() {
+        if(this.domesticConsent.getRisk()!=null){
+            if(this.domesticConsent.getRisk().getDeliveryAddress()!=null){
+                if(this.domesticConsent.getRisk().getDeliveryAddress().getCountrySubDivision()!=null && !this.domesticConsent.getRisk().getDeliveryAddress().getCountrySubDivision().isEmpty()){
+                    return this.domesticConsent.getRisk().getDeliveryAddress().getCountrySubDivision().get(0);
+                }
+            }
+        }
+        return null;
     }
```

Comment:
Obviously I prefer this approach (except for the multiple calls to `getRisk().getDeliveryAddress()`, the long line length and lack of formatting!) 😆😆